src/app/clients/actionBarModal/ShippingLabelsModal.tsx

```diff
@@ -0,0 +1,72 @@
+import React from "react";
+import styled from "styled-components";
+import ShippingsLabel from "@/pdf/ShippingLabels/ShippingLabel";
+import Modal from "@/components/Modal/Modal";
+import dayjs, { Dayjs } from "dayjs";
+import { ClientsTableRow } from "@/app/clients/getClientsTableData";
+
+interface SharedModalProps {
+    isOpen: boolean;
+    onClose: () => void;
+    data: ClientsTableRow[];
+    status: string | null;
+    onSubmit: (date: Dayjs) => void;
+    header: string;
+    headerId: string;
+    errorText: string | null;
+}
+
+const Centerer = styled.div`
+    display: flex;
+    justify-content: center;
+`;
+
+const ModalInner = styled.div`
+    display: flex;
+    flex-direction: column;
+    gap: 1rem;
+    align-items: stretch;
+`;
+
+const StatusText = styled.p`
+    margin-left: 1rem;
+    border-top: 1px solid darkgrey;
+    padding: 1rem 0;
+    &:last-child {
+        border-bottom: 1px solid darkgrey;
+    }
+`;
+
+const ShoppingLabelsModal: React.FC<SharedModalProps> = (props) => {
+    const parcelIds = props.data.map((parcel) => {
+        return parcel.parcelId;
+    });
+
+    return (
+        <Modal {...props} header={props.header} headerId={props.headerId}>
+            <ModalInner>
+                Parcels selected for printing:
+                <div>
+                    {props.data.map((parcel, index) => {
+                        return (
+                            <StatusText key={index}>
+                                {parcel.collectionCentre}
+                                {parcel.fullName && ` - ${parcel.fullName}`}
+                                {parcel.collectionDatetime &&
+                                    `\n @ ${dayjs(parcel.collectionDatetime!).format(
+                                        "DD/MM/YYYY HH:mm"
+                                    )}`}
+                            </StatusText>
+                        );
+                    })}
+                </div>
+                {props.errorText && <small>{props.errorText}</small>}
+                <Centerer>
+                    <ShippingsLabel text="Print" parcelIds={parcelIds} />
```

Comment:
Just some questions 
1. Would it be better to put the print button like on the top (e.g. upper right corner) so if there is like a super long list to print out people don't have to scroll to the most bottom - though idk if it would be then inconsistent to general modal theming with the button usually at the bottom
2. What is the max number of shipping label you have tried to print out - currently I tried like 28 and it already took a while to load (what if there is like hundreds or even 1000 (if there is like 1000 lines then people just check all and this could easily happen - and might crash the program?)